docs/shared/tutorials/gradle.md

```diff
@@ -0,0 +1,329 @@
+}
+
+dependencies {
+    implementation("org.apache.commons:commons-text")
+    implementation(project(":utilities"))
+}
+
+tasks.register("mytask"){
+    group = "Custom"
+    description = "A custom task"
+}
+
+tasks.named("mytask"){
+    doFirst {
+        println("This is my task!")
+    }
+}
+
+application {
+    // Define the main class for the application.
+    mainClass = "org.example.app.App"
+}
+```
+
+Now, you can run `mytask` with Nx like this:
+
+```{% command="./nx mytask app" %}
+> nx run app:mytask
+
+> /Users/isaac/Documents/code/gradle/gradlew mytask
+
+
+> Task :app:mytask
+This is my task!
+
+BUILD SUCCESSFUL in 522ms
+11 actionable tasks: 2 executed, 9 up-to-date
+```
+
+You can see all the tasks available to Nx by opening the project details view for the `app` project. You can do this either through [Nx Console](/getting-started/editor-setup) or from the terminal:
+
+```shell {% path="~/tuskydesigns" %}
+npx nx show project @tuskdesign/demo --web
+```
+
+{% project-details title="Project Details View" jsonFile="shared/tutorials/gradle-pdv.json" %}
+{% /project-details %}
+
+## Run Tasks for Affected Projects
+
+Nx doesn't just cache your task results, it can also eliminate the need to run unnecessary tasks. Nx can use its project graph in conjunction with your git history to only run tasks for projects that may have been affected by the changes that you made. Let's explore how the [nx affected](/ci/features/affected) command works.
+
+First, commit any outstanding changes in your repository:
+
+```shell
+git commit -am "changes"
+```
+
+Next make a small change to the `app` code:
+
+```java {% fileName="app/src/main/java/org/example/app/App.java" %}
+/*
+ * This source file was generated by the Gradle 'init' task
+ */
+package org.example.app;
+
+import org.example.list.LinkedList;
+
+import static org.example.utilities.StringUtils.join;
+import static org.example.utilities.StringUtils.split;
+import static org.example.app.MessageUtils.getMessage;
+
+import org.apache.commons.text.WordUtils;
+
+public class App {
+    public static void main(String[] args) {
+        LinkedList tokens;
+        tokens = split(getMessage());
+        String result = join(tokens) + " something";
+        System.out.println(WordUtils.capitalize(result));
+    }
+}
+```
+
+Then you can use the `nx affected` command to run the `assemble` task only for projects that were affected by that change.
+
+```shell
+./nx affected -t assemble
+```
+
+The assemble task was run for the `app` project, but not for `list` or `utilities` because a change to the `app` project could not have changed the behavior of those projects. This feature is most useful in CI where you want to make sure to fully check every project that may have been affected by a change, but you want to avoid wasting time on tasks that do not need to be run. As more projects are added to the repository, this functionality becomes essential to maintain a fast CI pipeline.
```

Comment:
```suggestion
The `assemble` task was run for the `app` project, but not for `list` or `utilities` because a change to the `app` project could not have changed the behavior of those projects. This feature is most useful in CI where you want to make sure to fully check every project that may have been affected by a change, but you want to avoid wasting time on tasks that do not need to be run. As more projects are added to the repository, this functionality becomes essential to maintain a fast CI pipeline.
```